ICSharpCode.Decompiler/CSharp/CallBuilder.cs

```diff
@@ -1057,30 +1046,46 @@ private bool TransformParamsArgument(ExpectedTargetDetails expectedTargetDetails
 			}
 			return false;
 
-			bool CheckArgument(out int len, out IType t)
+			bool ExtractArguments(out IType elementType, out List<IParameter> parameters, out List<TranslatedExpression> arguments)
 			{
-				len = 0;
-				t = null;
-				if (arg.ResolveResult is CSharpInvocationResolveResult csirr &&
-					csirr.Arguments.Count == 0 && csirr.Member is IMethod emptyMethod &&
-					emptyMethod.IsStatic &&
-					"System.Array.Empty" == emptyMethod.FullName &&
-					emptyMethod.TypeArguments.Count == 1)
+				elementType = null;
+				parameters = null;
+				arguments = null;
+				switch (paramsArgument.ResolveResult)
 				{
-					t = emptyMethod.TypeArguments[0];
-					return true;
-				}
-
-				if (arg.ResolveResult is ArrayCreateResolveResult acrr &&
-					acrr.SizeArguments.Count == 1 &&
-					acrr.SizeArguments[0].IsCompileTimeConstant &&
-					acrr.SizeArguments[0].ConstantValue is int l)
-				{
-					len = l;
-					t = ((ArrayType)acrr.Type).ElementType;
-					return true;
+					case CSharpInvocationResolveResult { Member: IMethod method, Arguments: var args }:
+						// match System.Array.Empty<T>()
+						if (args is [] && method is { IsStatic: true, FullName: "System.Array.Empty", TypeArguments: [var type] })
+						{
+							elementType = type;
+							arguments = new();
+							parameters = new();
+							return true;
+						}
+						if (paramsArgument.Expression is ObjectCreateExpression oce
+							&& method is { IsConstructor: true, DeclaringType: { TypeArguments: [var type2] } declaringType }
+							&& declaringType.IsKnownType(KnownTypeCode.ReadOnlySpanOfT))
```

Comment:
This matches every overload of `new ReadOnlySpan<T>`; but I think it's meant to only apply to the `(in T)` overload?